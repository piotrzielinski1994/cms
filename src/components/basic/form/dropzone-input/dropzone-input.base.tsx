import { EnhancedHtmlProps, HtmlProps } from '@/utils/html/html.types';
import { forwardRef } from 'react';
import Form from '../root/form';

// prettier-ignore
type InputProps = EnhancedHtmlProps<'input', {
  name: string;
  accept?: string;
  multiple?: boolean;
}>;

const Wrapper = (props: HtmlProps['label']) => {
  return <label {...props} />;
};

const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
  return <input type="file" ref={ref} {...props} />;
});

const Items = (props: HtmlProps['ul']) => {
  return <ul {...props} />;
};

const Item = (props: HtmlProps['li']) => {
  return <li {...props} />;
};

const ItemButton = (props: HtmlProps['button']) => {
  return <button type="button" {...props} />;
};

const Placeholder = (props: HtmlProps['li']) => {
  return <span {...props} />;
};

const DropzoneInputBase = {
  Root: Form.Group,
  Label: Form.Label,
  Wrapper,
  Input,
  Items,
  Item,
  Placeholder,
  ItemButton,
  Error: Form.Error,
};

export default DropzoneInputBase;
