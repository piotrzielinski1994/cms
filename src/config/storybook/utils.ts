import { ReactNode } from 'react';

const DEFAULT_VALUE = '__default__' as const;
const THUMBNAIL_ID = 'thumbnail';

const storyToUrlPath = (title: string) => {
  const parts = title.split('/');
  const story = parts[parts.length - 1];
  const titlePart = title.toLowerCase().replace(/\//g, '-').replace(/\s+/g, '-');
  const storyPart = story
    .replace(/([a-z0-9])([A-Z])/g, '$1-$2')
    .replace(/([a-zA-Z])(\d+)/g, '$1-$2')
    .toLowerCase();
  return `${titlePart}--${storyPart}`;
};

const getFallback = <T extends ReactNode>(value: T, translation: string): T => {
  return value === DEFAULT_VALUE ? (translation as T) : value;
};

export { DEFAULT_VALUE, getFallback, storyToUrlPath, THUMBNAIL_ID };
