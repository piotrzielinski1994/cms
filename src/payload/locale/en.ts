const en = {
  common: {
    content: 'Content',
    layout: 'Layout',
    component: {
      singular: 'Component',
      plural: 'Components',
    },
    section: {
      singular: 'Section',
      plural: 'Sections',
    },
    related: {
      singular: 'Related',
      plural: 'Related',
    },
    openInNewTab: 'Open in new tab',
    item: {
      singular: 'Item',
      plural: 'Items',
    },
    files: {
      singular: 'File',
      plural: 'Files',
    },
    default: 'Default',
    dark: 'Dark',
    question: 'Question',
    answer: 'Answer',
    pageNotFound: 'This page could not be found.',
    goHome: 'Go home',
  },
  fields: {
    label: 'Label',
    slug: 'Slug',
    path: 'Path',
    type: 'Type',
    title: 'Title',
    heading: 'Heading',
    subheading: 'Subheading',
    description: 'Description',
    image: 'Image',
    publishedAt: 'Published at',
    fullName: 'Full name',
    parent: 'Parent element',
    children: 'Child elements',
    role: {
      singular: 'Role',
      plural: 'Roles',
    },
    menu: 'Menu',
    submenu: 'Submenu',
    navItem: 'Navigation item',
    internalLink: 'Internal link',
    documentToLinkTo: 'Document to link to',
    buttons: 'Buttons',
    selector: 'Selector',
    isReversed: 'Reversed layout',
    subpages: 'Subpages',
    seo: 'SEO',
    acceptLabel: 'Accept label',
    readMoreLabel: 'Read more label',
    url: 'URL',
  },
  groups: {
    admin: 'Admin',
    docs: 'Docs',
  },
  collections: {
    pages: {
      singular: 'Page',
      plural: 'Pages',
    },
    categories: {
      singular: 'Category',
      plural: 'Categories',
    },
    posts: {
      singular: 'Post',
      plural: 'Posts',
    },
    sites: {
      singular: 'Site',
      plural: 'Sites',
    },
    users: {
      singular: 'User',
      plural: 'Users',
    },
    images: {
      singular: 'Image',
      plural: 'Images',
    },
    redirects: {
      singular: 'Redirect',
      plural: 'Redirects',
    },
  },
  enums: {
    role: {
      user: 'User',
      editor: 'Editor',
      admin: 'Admin',
    },
  },
  components: {
    header: 'Header',
    footer: 'Footer',
    cookiesBanner: 'Cookies banner',
  },
  frontend: {
    submit: 'Submit',
    logo: 'Logo',
    localeSwitcher: 'Locale switcher',
    themeSwitcher: 'Theme switcher',
    fontScaleSwitcher: 'Font scale switcher',
    increment: 'Increment',
    decrement: 'Decrement',
    close: 'Close',
    component: {
      skipLink: 'Go to the main content',
    },
    contactForm: {
      fields: {
        email: {
          label: 'Email',
        },
        message: {
          label: 'Message',
        },
      },
    },
  },
  zod: {
    invalid_type: 'Expected {expected}, received {received}',
    invalid_type_with_path: '{path} is expected {expected}, but received {received}',
    invalid_type_received_undefined: 'Required',
    invalid_literal: 'Invalid literal value, expected {expected}',
    unrecognized_keys: 'Unrecognized key(s) in object: {- keys}',
    unrecognized_keys_one: 'Unrecognized key in object: {- keys}',
    unrecognized_keys_other: 'Unrecognized keys in object: {- keys}',
    invalid_union: 'Invalid input',
    invalid_union_discriminator: 'Invalid discriminator value. Expected {- options}',
    invalid_enum_value: 'Invalid enum value. Expected {- options}, received {received}',
    invalid_arguments: 'Invalid function arguments',
    invalid_return_type: 'Invalid function return type',
    invalid_date: 'Invalid date',
    custom: 'Invalid input',
    invalid_intersection_types: 'Intersection results could not be merged',
    not_multiple_of: 'Number must be a multiple of {multipleOf}',
    not_finite: 'Number must be finite',
    invalid_string: {
      email: 'Invalid {validation}',
      url: 'Invalid {validation}',
      uuid: 'Invalid {validation}',
      cuid: 'Invalid {validation}',
      regex: 'Invalid',
      datetime: 'Invalid {validation}',
      startsWith: 'Invalid input: must start with {startsWith}',
      endsWith: 'Invalid input: must end with {endsWith}',
    },
    too_small: {
      array: {
        inclusive: 'Array must contain at least {minimum} element(s)',
        inclusive_one: 'Array must contain at least {minimum} element',
        inclusive_other: 'Array must contain at least {minimum} elements',
        not_inclusive: 'Array must contain more than {minimum} element(s)',
        not_inclusive_one: 'Array must contain more than {minimum} element',
        not_inclusive_other: 'Array must contain more than {minimum} elements',
      },
      string: {
        inclusive: 'String must contain at least {minimum} character(s)',
        inclusive_one: 'String must contain at least {minimum} character',
        inclusive_other: 'String must contain at least {minimum} characters',
        inclusive_with_path: '{path} must contain at least {minimum} character(s)',
        inclusive_with_path_one: '{path} must contain at least {minimum} character',
        inclusive_with_path_other: '{path} must contain at least {minimum} characters',
        not_inclusive: 'String must contain over {minimum} character(s)',
        not_inclusive_one: 'String must contain over {minimum} character',
        not_inclusive_other: 'String must contain over {minimum} characters',
        not_inclusive_with_path: '{path} must contain over {minimum} character(s)',
        not_inclusive_with_path_one: '{path} must contain over {minimum} character',
        not_inclusive_with_path_other: '{path} must contain over {minimum} characters',
      },
      number: {
        inclusive: 'Number must be greater than or equal to {minimum}',
        inclusive_with_path: '{path} must be greater than or equal to {minimum}',
        not_inclusive: 'Number must be greater than {minimum}',
        not_inclusive_with_path: '{path} must be greater than {minimum}',
      },
      set: {
        inclusive: 'Invalid input',
        not_inclusive: 'Invalid input',
      },
      date: {
        inclusive: 'Date must be greater than or equal to {- minimum, datetime}',
        not_inclusive: 'Date must be greater than {- minimum, datetime}',
      },
    },
    too_big: {
      array: {
        inclusive: 'Array must contain at most {maximum} element(s)',
        inclusive_one: 'Array must contain at most {maximum} element',
        inclusive_other: 'Array must contain at most {maximum} elements',
        not_inclusive: 'Array must contain less than {maximum} element(s)',
        not_inclusive_one: 'Array must contain less than {maximum} element',
        not_inclusive_other: 'Array must contain less than {maximum} elements',
      },
      string: {
        inclusive: 'String must contain at most {maximum} character(s)',
        inclusive_one: 'String must contain at most {maximum} character',
        inclusive_other: 'String must contain at most {maximum} characters',
        inclusive_with_path: '{path} must contain at most {maximum} character(s)',
        inclusive_with_path_one: '{path} must contain at most {maximum} character',
        inclusive_with_path_other: '{path} must contain at most {maximum} characters',
        not_inclusive: 'String must contain under {maximum} character(s)',
        not_inclusive_one: 'String must contain under {maximum} character',
        not_inclusive_other: 'String must contain under {maximum} characters',
        not_inclusive_with_path: '{path} must contain under {maximum} character(s)',
        not_inclusive_with_path_one: '{path} must contain under {maximum} character',
        not_inclusive_with_path_other: '{path} must contain under {maximum} characters',
      },
      number: {
        inclusive: 'Number must be less than or equal to {maximum}',
        inclusive_with_path: '{path} must be less than or equal to {maximum}',
        not_inclusive: 'Number must be less than {maximum}',
        not_inclusive_with_path: '{path} must be less than {maximum}',
      },
      set: {
        inclusive: 'Invalid input',
        not_inclusive: 'Invalid input',
      },
      date: {
        inclusive: 'Date must be smaller than or equal to {- maximum, datetime}',
        not_inclusive: 'Date must be smaller than {- maximum, datetime}',
      },
    },
  },
  storybook: {
    basic: {
      form: {
        textInput: {
          default: {
            label: 'Text Input',
            placeholder: 'Type value',
          },
          disabled: {
            label: 'Disabled Input',
            placeholder: 'Disabled',
          },
          invalid: {
            label: 'Invalid Input',
            value: 'Invalid value',
            error: 'Error message',
          },
        },
        numberInput: 'Number Input',
        textArea: 'Text Area',
        checkbox: {
          default: {
            label: 'Checkbox',
          },
          disabled: {
            label: 'Disabled Checkbox',
          },
          invalid: {
            label: 'Invalid Checkbox',
            error: 'Error message',
          },
        },
        radio: {
          default: {
            label: 'Radio Button',
          },
          disabled: {
            label: 'Disabled Radio Button',
          },
          invalid: {
            label: 'Invalid Radio Button',
            error: 'Error message',
          },
        },
        select: {
          label: 'Select',
          placeholder: 'Choose value',
          option: 'Option',
        },
      },
      button: {
        default: 'Button',
        disabled: 'Disabled',
        secondary: 'Secondary',
      },
      alert: {
        info: 'Information',
        success: 'Success',
        warn: 'Warning',
        error: 'Error',
      },
      badge: 'Badge',
      accordion: {
        heading: 'Heading',
        content: 'Content',
      },
      dialog: {
        button: 'Show Dialog',
        header: 'Header',
        content: 'Content',
        submit: 'Submit',
        cancel: 'Cancel',
      },
    },
  },
};

export { en };
