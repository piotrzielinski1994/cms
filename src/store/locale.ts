import { contentLocales, defaultContentLocale } from '@/config/locales.config';
import { Locale, useLocale } from 'next-intl';

const useLocaleStore = (): Locale => {
  const locale = useLocale();
  return contentLocales.includes(locale) ? locale : defaultContentLocale;
};

export { useLocaleStore };
