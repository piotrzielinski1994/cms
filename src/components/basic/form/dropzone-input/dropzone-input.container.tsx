import { useTranslations } from 'next-intl';
import { ComponentProps } from 'react';
import { DropzoneInput } from './dropzone-input';

const DropzoneInputContainer = (props: ComponentProps<typeof DropzoneInput>) => {
  const { t: translations, ...rest } = props;
  const t = useTranslations('frontend.component.uploadInput');
  const defaultTranslations = {
    clickToUpload: t('clickToUpload'),
    orDragAndDrop: t('orDragAndDrop'),
    fileExtensions: t('extensions.image'),
  };

  return <DropzoneInput t={{ ...defaultTranslations, ...translations }} {...rest} />;
};

export { DropzoneInputContainer };
