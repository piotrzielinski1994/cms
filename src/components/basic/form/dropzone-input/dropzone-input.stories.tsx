import Form from '@/components/basic/form/root/form';
import { DEFAULT_VALUE, getFallback } from '@/config/storybook/utils';
import type { Meta, StoryObj } from '@storybook/react';
import { useTranslations } from 'next-intl';
import { useState, type ComponentProps } from 'react';
import { DropzoneInput as DropzoneInputComponent } from './dropzone-input';

type Args = ComponentProps<typeof DropzoneInputComponent> & {
  label: string;
};

const meta: Meta<Args> = {
  component: DropzoneInputComponent,
  title: 'Basic/Form/DropzoneInput',
  argTypes: {
    label: { control: 'text' },
    multiple: { control: 'boolean' },
    disabled: { control: 'boolean' },
    fileNames: { table: { disable: true } },
    onFileRemove: { table: { disable: true } },
  },
  args: {
    label: DEFAULT_VALUE,
    multiple: true,
    disabled: false,
  },
};

const Render = ({ label, ...args }: Args) => {
  const t = useTranslations('storybook.basic.form');
  const [fileNames, setFileNames] = useState<string[]>([]);

  return (
    <div className="grid gap-2">
      <Form.Group>
        <Form.Label htmlFor="inpu1">{getFallback(label, t('dropzoneInput'))}</Form.Label>
        <DropzoneInputComponent
          {...args}
          id="inpu1"
          name="inpu1"
          fileNames={fileNames}
          onChange={(e) => {
            const { files } = e.target;
            if (!files) return;
            setFileNames((prev) => [...prev, ...[...files].map((file) => file.name)]);
          }}
          onFileRemove={(fileName) => {
            setFileNames((prev) => prev.filter((name) => name !== fileName));
          }}
        />
      </Form.Group>
      <Form.Group>
        <Form.Label htmlFor="input2">{t('textInput.disabled.label')}</Form.Label>
        <DropzoneInputComponent id="inpu2" name="inpu2" fileNames={[]} disabled />
      </Form.Group>
      <Form.Group>
        <Form.Label htmlFor="input3">{t('textInput.invalid.label')}</Form.Label>
        <DropzoneInputComponent
          id="inpu3"
          name="inpu3"
          fileNames={[]}
          error={t('textInput.invalid.error')}
        />
      </Form.Group>
    </div>
  );
};

const DropzoneInput: StoryObj<typeof DropzoneInputComponent> = { render: Render };

export { DropzoneInput };
export default meta;
