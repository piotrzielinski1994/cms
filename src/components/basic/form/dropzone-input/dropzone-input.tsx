'use client';

import { cn } from '@/utils/tailwind';
import { Upload, X } from 'lucide-react';
import { useTranslations } from 'next-intl';
import { ChangeEvent, InputHTMLAttributes, useState } from 'react';

type DropzoneInputProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> & {
  accept?: string;
  multiple?: boolean;
  fileNames: string[];
  onFileRemove?: (fileName: string) => void;
};

const DropzoneInput = ({ fileNames, onFileRemove, ...props }: DropzoneInputProps) => {
  const t = useTranslations('frontend.component.uploadInput');
  const [isDragging, setIsDragging] = useState(false);

  return (
    <>
      <label
        className={cn(
          'p-4 bg-red-500',
          'border border-solid border-foreground',
          'bg-input text-foreground/50',
          'has-[:enabled]:hover:border-foreground/90',
          'tw-has-focus:tw-cms-outline',
          { 'bg-foreground/5': isDragging },
          'grid gap-4',
          'cursor-pointer',
        )}
        onDragLeave={() => setIsDragging(false)}
        onDragOver={(e) => {
          e.preventDefault();
          setIsDragging(true);
        }}
        onDrop={(e) => {
          e.preventDefault();
          setIsDragging(false);
          const { files } = e.dataTransfer;
          const target = Object.assign(e.target, { files });
          const currentTarget = Object.assign(e.currentTarget, { files });
          const newEvent = Object.assign({}, e, { target, currentTarget });
          props.onChange?.(newEvent as unknown as ChangeEvent<HTMLInputElement>);
        }}
      >
        <div className="grid justify-items-center">
          <Upload />
          <p className="">
            <strong>{t('clickToUpload')}</strong> <span>{t('orDragAndDrop')}</span>
          </p>
          <p className="">{t('extensions.image')}</p>
          <input {...props} type="file" className={cn('sr-only', props.className)} />
        </div>
        {fileNames.length > 0 && (
          <ul
            role="group"
            className="grid gap-1"
            onKeyDown={(e) => {
              const arrowKeys = ['ArrowDown', 'ArrowRight', 'ArrowUp', 'ArrowLeft'];
              if (!arrowKeys.includes(e.key)) return;
              e.preventDefault();

              const buttons = Array.from(e.currentTarget.querySelectorAll('button'));
              const index = buttons.indexOf(document.activeElement as HTMLButtonElement);
              const offset = ['ArrowDown', 'ArrowRight'].includes(e.key) ? 1 : -1;

              buttons[(index + offset + buttons.length) % buttons.length]?.focus();
            }}
          >
            {fileNames.map((fileName, index) => (
              <li key={index}>
                <button
                  type="button"
                  tabIndex={index === 0 ? 0 : -1}
                  onClick={() => onFileRemove?.(fileName)}
                  className={cn(
                    'w-full p-2',
                    'grid grid-cols-[1fr_auto] gap-2',
                    'border border-foreground',
                    'bg-background text-foreground',
                    'text-xs text-left',
                    'hover:bg-foreground/5',
                  )}
                >
                  <span>{fileName}</span>
                  <X height="1lh" width="auto" />
                </button>
              </li>
            ))}
          </ul>
        )}
      </label>
    </>
  );
};

export { DropzoneInput };
