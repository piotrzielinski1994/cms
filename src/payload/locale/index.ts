import { clientEnv } from '@/env.client';
import {
  DefaultTranslationKeys,
  NestedKeysStripped,
  SupportedLanguages,
  TFunction,
} from '@payloadcms/translations';
import { en } from '@payloadcms/translations/languages/en';
import { pl } from '@payloadcms/translations/languages/pl';
import { pick } from 'ramda';
import { en as customEn } from './en';
import { pl as customPl } from './pl';

// Types ====================================
export type AdminTranslations = TFunction<
  NestedKeysStripped<typeof customEn> | DefaultTranslationKeys
>;

// Variables ====================================
export const adminLocale = {
  list: pick(clientEnv.feature.locale.admin.list, { en, pl } as SupportedLanguages),
  customList: {
    en: customEn,
    pl: customPl,
  },
  default: clientEnv.feature.locale.admin.default,
};

export const contentLocale = {
  list: clientEnv.feature.locale.content.list,
  default: clientEnv.feature.locale.content.default,
};
