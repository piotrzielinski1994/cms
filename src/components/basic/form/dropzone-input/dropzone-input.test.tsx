import { en } from '@/payload/locale/en';
import { withProviders } from '@/utils/tests';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ComponentProps } from 'react';
import { describe, expect, it, vi } from 'vitest';
import { axe } from 'vitest-axe';
import { DropzoneInput } from './dropzone-input';

const t = en.frontend.component.uploadInput;

describe('DropzoneInput', () => {
  const defaultProps = {
    name: 'dropzone-input',
    fileNames: ['file-1.webp', 'file-2.webp'],
    onFileRemove: vi.fn(),
    onChange: vi.fn(),
    t: {
      clickToUpload: t.clickToUpload,
      orDragAndDrop: t.orDragAndDrop,
      fileExtensions: t.extensions.image,
    },
  } satisfies ComponentProps<typeof DropzoneInput>;

  it('should have no accessibility violations', async () => {
    const { container } = render(withProviders(<DropzoneInput {...defaultProps} />));
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', () => {
    const { container } = render(withProviders(<DropzoneInput {...defaultProps} />));
    expect(container).toMatchSnapshot();
  });

  describe('Interactivity', () => {
    it('should call onChange when file is dropped', async () => {
      const newFileName = 'file-3.webp';
      const { getByText } = render(withProviders(<DropzoneInput {...defaultProps} />));

      const input = getByText(t.clickToUpload)
        .closest('label')!
        .querySelector('input') as HTMLInputElement;
      const label = input.closest('label')!;
      const file = new File(['file contents'], newFileName, { type: 'image/webp' });

      await userEvent.upload(input, file);

      const dataTransfer = new DataTransfer();
      dataTransfer.items.add(file);

      await userEvent.pointer([{ target: label, keys: '[DragEnter]' }]);
      label.dispatchEvent(
        new DragEvent('drop', {
          bubbles: true,
          cancelable: true,
          dataTransfer,
        }),
      );

      const calledEvent = defaultProps.onChange.mock.calls[0][0];
      expect(calledEvent.target.files![0].name).toBe(newFileName);
      expect(defaultProps.onChange).toHaveBeenCalled();
    });

    it('should call onFileRemove when a button is clicked', async () => {
      const { getByRole } = render(withProviders(<DropzoneInput {...defaultProps} />));
      const button = getByRole('button', { name: defaultProps.fileNames[1] });

      await userEvent.click(button);

      expect(defaultProps.onFileRemove).toHaveBeenCalledWith(defaultProps.fileNames[1], 1);
    });
  });

  describe('Keyboard', () => {
    it('should move focus with arrow keys', async () => {
      const { getByText, getAllByRole } = render(
        withProviders(<DropzoneInput {...defaultProps} />),
      );
      const input = getByText(t.clickToUpload).closest('label')!.querySelector('input');
      const buttons = getAllByRole('button');

      // Focus input
      await userEvent.tab();
      expect(document.activeElement).toBe(input);

      // Navigate across buttons (files)
      await userEvent.tab();
      expect(document.activeElement).toBe(buttons[0]);

      await userEvent.keyboard('{ArrowDown}');
      expect(document.activeElement).toBe(buttons[1]);

      await userEvent.keyboard('{ArrowUp}');
      expect(document.activeElement).toBe(buttons[0]);

      await userEvent.keyboard('{ArrowRight}');
      expect(document.activeElement).toBe(buttons[1]);

      await userEvent.keyboard('{ArrowLeft}');
      expect(document.activeElement).toBe(buttons[0]);

      // Move out of component
      await userEvent.tab();
      expect(document.activeElement).toBe(document.body);
    });
  });
});
