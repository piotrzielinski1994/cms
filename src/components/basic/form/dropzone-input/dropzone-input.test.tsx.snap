// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropzoneInput > should match the snapshot 1`] = `
<div>
  <div
    class="grid gap-1 content-start grid-rows-[auto_1fr]"
  >
    <label
      class="p-4 border border-solid border-foreground bg-input text-foreground/50 has-[:enabled]:hover:border-foreground/90 tw-has-focus:tw-cms-outline grid gap-4 cursor-pointer has-[:disabled]:cursor-not-allowed has-[:disabled]:border-foreground/50"
    >
      <div
        class="grid justify-items-center"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-upload"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3v12"
          />
          <path
            d="m17 8-5-5-5 5"
          />
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
        </svg>
        <p>
          <strong>
            Click to upload
          </strong>
           
          <span>
            or drag and drop
          </span>
        </p>
        <p>
          JPG, PNG or WEBP
        </p>
        <input
          class="sr-only"
          name="dropzone-input"
          type="file"
        />
      </div>
      <ul
        class="grid gap-1"
      >
        <li>
          <button
            class="w-full p-2 grid grid-cols-[1fr_auto] gap-2 border border-current bg-background text-foreground text-xs text-left enabled:hover:bg-foreground/5 disabled:cursor-not-allowed disabled:text-foreground/50"
            tabindex="0"
            type="button"
          >
            <span>
              file-1.webp
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-x"
              fill="none"
              height="1lh"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="auto"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            class="w-full p-2 grid grid-cols-[1fr_auto] gap-2 border border-current bg-background text-foreground text-xs text-left enabled:hover:bg-foreground/5 disabled:cursor-not-allowed disabled:text-foreground/50"
            tabindex="-1"
            type="button"
          >
            <span>
              file-2.webp
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-x"
              fill="none"
              height="1lh"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="auto"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </button>
        </li>
      </ul>
    </label>
    <span
      aria-hidden="true"
      class="min-h-[1em] text-sm text-red-500 leading-none"
      role="alert"
    />
  </div>
</div>
`;
